python/tank/descriptor/bundle_cache_usage/writer_sqlite.py

```diff
@@ -0,0 +1,345 @@
+        self._db_connection.text_factory = str
+
+    def _execute(self, sql_statement, tuple=None):
+        """
+        Connects the database if not already connected and execute the
+        specified SQL statement.
+
+        :param sql_statement:
+        :param tuple:
+        :return:
+        """
+        if tuple:
+            return self._get_cursor().execute(sql_statement, tuple)
+        else:
+            return self._get_cursor().execute(sql_statement)
+
+    def _create_main_table(self):
+        """
+        Create the database main table if it doesn't exists.
+
+        NOTE:
+        SQLite does not have a storage class set aside for storing dates and/or times.
+        Instead, the built-in Date And Time Functions of SQLite are capable of storing
+        dates and times as TEXT, REAL, or INTEGER values:
+
+        Reference:
+        https://sqlite.org/datatype3.html
+        """
+        sql_statement = \
+            """CREATE TABLE IF NOT EXISTS %s ( 
+                %s integer PRIMARY KEY,
+                %s text NOT NULL UNIQUE,
+                %s integer,
+                %s integer,
+                %s integer
+            );""" % (
+                BundleCacheUsageSQLiteWriter.DB_MAIN_TABLE_NAME,
+                BundleCacheUsageSQLiteWriter.DB_COL_ID,
+                BundleCacheUsageSQLiteWriter.DB_COL_PATH,
+                BundleCacheUsageSQLiteWriter.DB_COL_ADD_DATE,
+                BundleCacheUsageSQLiteWriter.DB_COL_LAST_ACCESS_DATE,
+                BundleCacheUsageSQLiteWriter.DB_COL_ACCESS_COUNT
+            )
+        self._execute(sql_statement)
+
+    def _connect(self):
+        """
+        Open or re-open a connection to the database and returns it's connected object.
+        The method simply returns the existing connected if database is alreayd opened.
+
+        :return: A `sqlite3.Connection` object.
+        """
+        if self._db_connection is None:
+            self._db_connection = sqlite3.connect(self.path)
+            log.debug_db_inst("connected: %s" % (self.path))
+
+        return self._db_connection
+
+    def _create_bundle_entry(self, bundle_path, timestamp, initial_access_count):
+        sql_statement = """INSERT INTO %s(%s, %s, %s, %s) VALUES(?,?,?,?)""" % (
+            BundleCacheUsageSQLiteWriter.DB_MAIN_TABLE_NAME,
+            BundleCacheUsageSQLiteWriter.DB_COL_PATH,
+            BundleCacheUsageSQLiteWriter.DB_COL_ADD_DATE,
+            BundleCacheUsageSQLiteWriter.DB_COL_LAST_ACCESS_DATE,
+            BundleCacheUsageSQLiteWriter.DB_COL_ACCESS_COUNT
+        )
+
+        """ Connects and execute some SQL statement"""
+        bundle_entry_tuple = (bundle_path, timestamp, timestamp, initial_access_count)
+        self._execute(sql_statement, bundle_entry_tuple)
+
+    def _delete_bundle_entry(self, bundle_path):
+        'DELETE FROM tasks WHERE id=?'
+        sql_statement = "DELETE FROM %s WHERE %s=?" % (
+            BundleCacheUsageSQLiteWriter.DB_MAIN_TABLE_NAME,
```

Comment:
these constants don't help readability. i think better to just write out the sql and skip the constants. Kiss.